src/js/media-controller.js

```diff
@@ -730,67 +730,14 @@ class MediaController extends MediaContainer {
       this._pipUnavailable,
     );
 
-    // TODO: Update to propagate all states when registered
     if (this.media) {
-      propagateMediaState(
-        [el],
-        MediaUIAttributes.MEDIA_CAPTIONS_LIST,
-        stringifyTextTrackList(getCaptionTracks(this)) || undefined
-      );
-      propagateMediaState(
-        [el],
-        MediaUIAttributes.MEDIA_SUBTITLES_LIST,
-        stringifyTextTrackList(getSubtitleTracks(this)) || undefined
-      );
-      propagateMediaState(
-        [el],
-        MediaUIAttributes.MEDIA_CAPTIONS_SHOWING,
-        stringifyTextTrackList(getShowingCaptionTracks(this)) || undefined
-      );
-      propagateMediaState(
-        [el],
-        MediaUIAttributes.MEDIA_SUBTITLES_SHOWING,
-        stringifyTextTrackList(getShowingSubtitleTracks(this)) || undefined
-      );
-      propagateMediaState([el], MediaUIAttributes.MEDIA_PAUSED, getPaused(this));
-      // propagateMediaState([el], MediaUIAttributes.MEDIA_VOLUME_LEVEL, level);
-      propagateMediaState([el], MediaUIAttributes.MEDIA_MUTED, getMuted(this));
-      propagateMediaState([el], MediaUIAttributes.MEDIA_VOLUME, getVolume(this));
-      propagateMediaState(
-        [el],
-        MediaUIAttributes.MEDIA_VOLUME_LEVEL,
-        getVolumeLevel(this)
-      );
-      propagateMediaState(
-        [el],
-        MediaUIAttributes.MEDIA_IS_FULLSCREEN,
-        this.hasAttribute(MediaUIAttributes.MEDIA_IS_FULLSCREEN)
-      );
-      propagateMediaState(
-        [el],
-        MediaUIAttributes.MEDIA_IS_CASTING,
-        this.hasAttribute(MediaUIAttributes.MEDIA_IS_CASTING)
-      );
-      propagateMediaState(
-        [el],
-        MediaUIAttributes.MEDIA_CURRENT_TIME,
-        getCurrentTime(this)
-      );
-      propagateMediaState(
-        [el],
-        MediaUIAttributes.MEDIA_DURATION,
-        getDuration(this)
-      );
-      propagateMediaState(
-        [el],
-        MediaUIAttributes.MEDIA_SEEKABLE,
-        getSeekable(this)?.join(':')
-      );
-      propagateMediaState(
-        [el],
-        MediaUIAttributes.MEDIA_PLAYBACK_RATE,
-        getPlaybackRate(this)
-      );
+      Object.keys(MediaUIAttributes).forEach((attribute) => {
```

Comment:
Minor nit: we're now double propagating all the attrs like MEDIA_PIP_UNAVAILABLE (above this diff).

Reply:
Good call. I guess I should skip those 🤔